js/imageeditor/shared/tools/Tools.svelte

```diff
@@ -0,0 +1,160 @@
+	type transform_tool = "crop" | "rotate";
+	type brush_tool = "brush_color" | "brush_size";
+	type eraser_tool = "eraser_size";
+
+	export interface ToolOptions {
+		order: number;
+		label: string;
+		icon: typeof Image;
+		cb: (...args: any[]) => void;
+		id: upload_tool | transform_tool | brush_tool | eraser_tool;
+	}
+
+	export interface ToolMeta {
+		default: upload_tool | transform_tool | brush_tool | eraser_tool | null;
+		options: ToolOptions[];
+	}
+
+	export interface ToolContext {
+		register_tool: (type: tool, meta: ToolMeta) => () => void;
+		active_tool: {
+			subscribe(
+				this: void,
+				run: Subscriber<tool | null>,
+				invalidate?: Invalidator<tool | null>
+			): Unsubscriber;
+		};
+		activate_subtool: (
+			sub_tool: upload_tool | transform_tool | brush_tool | eraser_tool | null
+		) => void;
+	}
+</script>
+
+<script lang="ts">
+	import { Toolbar, IconButton } from "@gradio/atoms";
+	import { getContext, setContext } from "svelte";
+	import { writable } from "svelte/store";
+	import { EDITOR_KEY, type EditorContext } from "../ImageEditor.svelte";
+	import { Image, Crop, Brush, Erase } from "@gradio/icons";
+	import { type I18nFormatter } from "@gradio/utils";
+
+	const { pixi, current_layer, dimensions } =
+		getContext<EditorContext>(EDITOR_KEY);
+
+	export let i18n: I18nFormatter;
+
+	let tools: tool[] = [];
+	// export let transforms: ("crop" | "rotate")[];
+	// export let brush: Brush;
+	// export let eraser;
+
+	const active_tool = writable<tool | null>("bg");
+	const metas: Record<tool, ToolMeta | null> = {
+		draw: null,
+		erase: null,
+		crop: null,
+		bg: null
+	};
+
+	$: sub_menu = $active_tool && metas[$active_tool];
+	let sub_tool: upload_tool | transform_tool | brush_tool | eraser_tool | null;
+	const tool_context: ToolContext = {
+		register_tool: (type: tool, meta: ToolMeta) => {
+			tools = [...tools, type];
+			metas[type] = meta;
+
+			return () => {
+				tools = tools.filter((tool) => tool !== type);
+			};
+		},
+
+		active_tool: {
+			subscribe: active_tool.subscribe
+		},
+
+		activate_subtool: (
+			_sub_tool: upload_tool | transform_tool | brush_tool | eraser_tool | null
+		) => {
+			sub_tool = _sub_tool;
+		}
+	};
+
+	setContext<ToolContext>(TOOL_KEY, tool_context);
+
+	const tools_meta: Record<
+		tool,
+		{
+			order: number;
+			label: string;
+			icon: typeof Image;
+		}
+	> = {
+		bg: {
+			order: 0,
+			label: i18n("Image"),
+			icon: Image
+		},
+		crop: {
+			order: 1,
+			label: i18n("Transform"),
+			icon: Crop
+		},
+		draw: {
+			order: 2,
+			label: i18n("Draw"),
+			icon: Brush
+		},
+		erase: {
+			order: 2,
+			label: i18n("Draw"),
+			icon: Erase
+		}
+	} as const;
+</script>
+
+<slot />
+
+<div class="toolbar-wrap">
+	<Toolbar show_border={false}>
+		{#if sub_menu}
+			{#each sub_menu.options as meta (meta.id)}
+				<IconButton
+					highlight={sub_tool === meta.id}
+					on:click={meta.cb}
+					Icon={meta.icon}
+					size="large"
+					padded={false}
+				/>
```

Comment:
We need to add the concept of a text overlay here so we can display the current brush size over the icon. The text will probably need to be in a store so we can make sure it always stays up to date. 

This is arguably the worst part of the code atm. Svelte 5 is gonna make this a million times nicer.